src/api/pokemon.ts

```diff
@@ -0,0 +1,15 @@
+import {PokemonClient} from 'pokenode-ts';
+
+// getBerry returns berry info
+// TODO currently just prints stuff to console
+export const getPokemonList = async () => {
```

Comment:
we could add the return type for this function which seems like its `void | NamedAPIResource[] | APIResource[]`

side question: do we know why it can return either a list of `NamedApiResource` objects or a list of `ApiResource` objects? Do we need to handle those cases somehow?

```suggestion
export const getPokemonList = async (): Promise<void | NamedAPIResource[] | APIResource[]> => {
```

that type is pretty crazy though so could add an alias for it

```ts
type PokemonListPromise = Promise<void | NamedAPIResource[] | APIResource[]>
```

and then do

```suggestion
export const getPokemonList = async (): PokemonListPromise => {
```

if we go that route we might want a `types.ts` file so that our components have access to those types.

Reply:
I am not sure. The docs only specify that `listPokemons` returns a `NamedAPIResource[]`. All we would have to do is check if the name key was in the object.